File: leetcode-04/solutions/python3/0060.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        ans = ''
        nums = [i + 1 for i in range(n)]
        fact = [1] * n

        for i in range(1, n):
            fact[i] = fact[i - 1] * i

        k -= 1

        for i in range(n, 0, -1):
            j = k // fact[i - 1]
            k %= fact[i - 1]
            ans += str(nums[j])
            nums.pop(j)

        return ans

File: leetcode-04/solutions/cpp/0060.cpp
class Solution {
 public:
  string getPermutation(int n, int k) {
    string ans;
    vector<int> nums(n);
    vector<int> fact(n, 1);

    for (int i = 0; i < n; i++) nums[i] = i + 1;
    for (int i = 1; i < n; i++) fact[i] = fact[i - 1] * i;

    k--;

    for (int i = n; i >= 1; i--) {
      int j = k / fact[i - 1];
      k %= fact[i - 1];
      ans.append(to_string(nums[j]));
      nums.erase(nums.begin() + j);
    }

    return ans;
  }
};